v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/dialectadapter/postgresql/PostgreSQLDialectAdapter.java

```diff
@@ -413,15 +413,16 @@ public String getReadQuery(String tableName, ImmutableList<String> partitionColu
    * @param tableName name of the table to read.
    * @param partitionColumns partition columns.
    * @param timeoutMillis timeout of the count query in milliseconds. Set to 0 to disable timeout.
+   *     Note that PG does not have an easy way of adding a server level timeout hint in the single
+   *     statement. The client side prepared statement timeout which is set by {@link
+   *     com.google.cloud.teleport.v2.source.reader.io.jdbc.uniformsplitter.transforms.ReadWithUniformPartitions
+   *     ReadWithUniformPartitions} will help in capping the time query spends in counting the rows.
    * @return Query Statement.
    */
   @Override
   public String getCountQuery(
       String tableName, ImmutableList<String> partitionColumns, long timeoutMillis) {
-    return addWhereClause(
```

Comment:
Which bit of this PR fixes the "No results were returned by the query" error ?

All code changes seem related to the uniformization hints. 

Maybe a short description in the PR on what is helping fix the error would help.

Reply:
`No results were returned by the query` was caused by `SET statement_timeout = %s;` part of the query.
  As we have added in the code comments,  PG does not have a way to add statement level timeout (via an inline hint) like MySql. We also have a JDBC level timeout enforced by Java, so the timeout handling will get taken care of. I will catchup with you separately to discuss this change in detail.